From e02518f80df01228b45d34bdf175a02bfe962d67 Mon Sep 17 00:00:00 2001
From: Prasanth Jayachandran <prasanthj@apache.org>
Date: Fri, 9 Dec 2016 14:55:10 -0800
Subject: [PATCH 1348/1431] CDH-62829: HIVE-15329: NullPointerException might
 occur when create table (Meilong Huang reviewed
 by Prasanth Jayachandran)

(cherry picked from commit 88e86b98835b949f8410259c6dab0c802233bb73)

Change-Id: Iddbca0908551c79b4e2e0444270a33adfbeaa466
(cherry picked from commit acd36140a5a0619d85a972c7cf5cffbcf90965a4)
---
 .../hadoop/hive/metastore/MetaStoreUtils.java      |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreUtils.java b/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreUtils.java
index 6fd244d..5062f77 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreUtils.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreUtils.java
@@ -1388,7 +1388,7 @@ public static Path getOriginalLocation(
   }
 
   public static boolean isNonNativeTable(Table table) {
-    if (table == null) {
+    if (table == null || table.getParameters() == null) {
       return false;
     }
     return (table.getParameters().get(hive_metastoreConstants.META_TABLE_STORAGE) != null);
-- 
1.7.9.5

